backend/api/Services/InspectionService.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Linq.Dynamic.Core;
+using Api.Controllers.Models;
+using Api.Database.Context;
+using Api.Database.Models;
+using Api.Utilities;
+using Microsoft.EntityFrameworkCore;
+namespace Api.Services
+{
+    public interface IInspectionService
+    {
+        public Task<Inspection?> AddFindings(InspectionFindingsQuery inspectionFindingsQuery);
+        public Task<Inspection?> ReadById(string id);
+        public Task<MissionRun?> GetMissionRunFromIsarStepId(string isarStepId);
+    }
+
+    [SuppressMessage(
+        "Globalization",
+        "CA1309:Use ordinal StringComparison",
+        Justification = "EF Core refrains from translating string comparison overloads to SQL"
+    )]
+    [SuppressMessage(
+        "Globalization",
+        "CA1304:Specify CultureInfo",
+        Justification = "Entity framework does not support translating culture info to SQL calls"
+    )]
+    public class InspectionService : IInspectionService
+    {
+        private readonly FlotillaDbContext _context;
+
+        public InspectionService(
+            FlotillaDbContext context)
+        {
+            _context = context; ;
+        }
+
+        public async Task<Inspection?> AddFindings(InspectionFindingsQuery inspectionFindingsQuery)
+        {
+
+            var missionRun = await GetMissionRunFromIsarStepId(inspectionFindingsQuery.IsarStepId);
+
+            if (missionRun != null) //missionRun == null
+            {
+                var inspection = missionRun.Tasks
+                    .SelectMany(task => task.Inspections)
+                        .FirstOrDefault(i => i.IsarStepId == inspectionFindingsQuery.IsarStepId);
+
+                if (inspection != null)
+                {
+                    // Update the inspection with the new values
+                    var findings = new InspectionFindings(inspectionFindingsQuery);
+                    inspection.InspectionFindings.Add(findings);
+                    _context.MissionRuns.Update(missionRun);
+                    // Save the changes to the database
+                    await _context.SaveChangesAsync();
+
+                    return inspection;
+                }
+                else
+                {
+                    return null;
+                }
```

Comment:
```suggestion
                else { return null; }
```

Reply:
I believe this is addressed now, hence resolving.